modules/pel/peltool/user_data.py

```diff
@@ -0,0 +1,81 @@
+from pel.datastream import DataStream
+from collections import OrderedDict
+from pel.peltool.pel_values import *
+from enum import Enum, unique
+import json
+from
+
+
+@unique
+class UserDataFormat(Enum):
+    json = 0x1
+    cbor = 0x2
+    text = 0x3
+    custom = 0x4
+
+
+def get_value(data: memoryview, start: int, end: int) -> int:
+    return int.from_bytes(data[start: start + end], byteorder="big")
+
+
+class UserData:
+    """
+    This represents the User Data section in a PEL.  It is free form data
+    that the creator knows the contents of.  The component ID, version,
+    and sub-type fields in the section header are used to identify the
+    format.
+
+    The Section base class handles the section header structure that every
+    PEL section has at offset zero.
+    """
+
+    def __init__(self, stream: DataStream, sectionID: int, sectionLen: int,
+                 versionID: int, subType: int, componentID: int, creatorID: str):
+        self.stream = stream
+        self.sectionID = sectionID
+        self.sectionLen = sectionLen
+        self.versionID = versionID
+        self.subType = subType
+        self.componentID = componentID
+        self.creatorID = creatorID
+        self.dataLength = sectionLen - 8
+        self.data = self.stream.get_mem(self.dataLength)
+
+    def parse(self, out: OrderedDict) -> str:
+        import importlib
+        name = (self.creatorID.lower() + "%04X" % self.componentID).lower()
+        try:
+            cls = importlib.import_module("udparsers." + name + "." + name)
+            mv = memoryview(self.data)
+            return cls.parseUDToJson(self.subType, self.versionID, mv)
+        except:
+            from udparsers.helpers.errludP_Helpers import hexDump
+            out["Data"] = hexDump(self.data, 0, self.dataLength)
```

Comment:
ok, I see why I was confused.  your commit seems to be deleting the existing hexdump.py file?

so there is a local hex dump you can use: https://github.com/ibm-openbmc/openpower-pel-parsers/blob/master/modules/pel/hexdump.py

Reply:
> ok, I see why I was confused. your commit seems to be deleting the existing hexdump.py file?
> 
> so there is a local hex dump you can use: https://github.com/ibm-openbmc/openpower-pel-parsers/blob/master/modules/pel/hexdump.py

Oh, Sorry, Maybe I made a mistake :(
Restored